src/ess/dream/workflow.py

```diff
@@ -37,13 +40,14 @@ def default_parameters() -> dict:
     }
 
 
-def DreamGeant4Workflow() -> sciline.Pipeline:
+def DreamGeant4Workflow(*, run_norm: RunNormalization) -> sciline.Pipeline:
     """
     Workflow with default parameters for the Dream Geant4 simulation.
     """
     wf = LoadGeant4Workflow()
     for provider in itertools.chain(powder_providers, _dream_providers):
         wf.insert(provider)
+    run_norm.insert(wf)
```

Comment:
I was quite confused by this interface. I actually thought you got the order wrong, and that it should have been `wf.insert(run_norm)`.
Then I saw that you added a `insert` method to the `Enum`.

I was expecting this to work more like e.g. the `UncertaintyBroadcastMode`, which is just a parameter we set on the workflow (e.g. https://github.com/scipp/esssans/blob/main/docs/user-guide/isis/sans2d.ipynb?short_path=e98bbec#L141)?

Was there a reason I'm missing that meant you chose this interface?

Reply:
1. I used `run_norm.insert(wf)` instead of `wf.insert(run_norm)` because I thought it would be good to keep the selection logic within `RunNormalization`. I see how the call can be confusing. I can change it to a free function. But I don't want the `match` statement inlined into `DreamGeant4Workflow`.
2. We cannot use a parameter like `UncertaintyBroadcastMode` because the different normalizations need to request different data from the pipeline. Using a combined providers that picks a norm would require requesting the union of all dependencies, that is monitors *and* proton charge.

Reply:
I think it would be better to have a free function and remove it from the Enum?

An alternative could be to make classmethods that would generate the workflow with the correct providers?
e.g.
```Py
LoadGeant4Workflow.normalized_by_monitor_histogram()
LoadGeant4Workflow.normalized_by_monitor_integrated()
LoadGeant4Workflow.normalized_by_proton_charge()
```
and one of them could be the default?

Reply:
Ah yes, I forgot that it wasn't a class :roll_eyes:

Reply:
Done